### ISSUE TITLE
MMOItems integration

### ISSUE THREAD

**Issue by ArcticRaven** [2008Choco/Alchema#9]
**Describe the new feature**
MMOItems integration to allow the creation/usage of MMOItems.

**How will it work?**
Like witchcraft! But seriously - similar to the 'mincraft:potato' or 'alchema:waffles' we could just have 'mmoitem:chickennuggies' or something. Issues may stem from MMOItems have both a name and a type, not sure how that would be handled. 

**Does this require integration with another plugin?**
Yes - and unfortunately the resource is premium, however I am more than willing to provide a test environment. API information can be found [here](https://git.lumine.io/mythiccraft/mmoitems/-/wikis/Main%20API%20Features)

**Comment by 2008Choco** [comment]
Looks like MMO items does have a relatively decent API to fetch items. I think optional integration with this plugin is definitely viable and I want to integrate natively with more plugins like this because they fit well with servers that would be running Alchema. I may even look into integrating this today 😄

I'm not too concerned about premium status, I'm a moderator on SpigotMC and can download premium resources free of charge. I don't, however, see a Maven repository which is going to make things painful for me. Compilation would require local installation of the plugin. I'll join the Discord and see if there are ways to depend on it.

**TL;DR:** Yes! I'd love to. Depends on whether or not I can depend on it from Maven remotely.

**EDIT:** Found a Maven repository I can properly depend on! Will work on integration today.

**Comment by 2008Choco** [comment]
```json
{
    "name": "MMOItems Gemstone",
    "description": "A recipe to test integration with MMOItems",
    "result": {
        "item": "minecraft:emerald",
        "name": "&aMMOItems Gemstone"
    },
    "ingredients": [
        {
            "type": "mmoitems:mmoitem",
            "item_type": "SWORD",
            "id": "CUTLASS"
        },
        {
            "type": "alchema:material",
            "item": "minecraft:emerald"
        }
    ]
}
```
The above is now supported and will be available in a future version 😄

**Comment by ArcticRaven** [comment]
Built Alchema current build from URL through Intellij - your provided template worked fine - however MMOItems don't seem to able to be used as a recipe result.
